gamemode/core/libs/sh_chatbox.lua

```diff
@@ -3,22 +3,33 @@ nut.chat.classes = nut.char.classes or {}
 
 local DUMMY_COMMAND = {syntax = "<string text>", onRun = function() end}
 
-if (!nut.command) then
+if (not nut.command) then
 	include("sh_command.lua")
 end
 
 -- Returns a timestamp
 function nut.chat.timestamp(ooc)
-	return nut.config.get("chatShowTime") and (ooc and " " or "").."("..nut.date.getFormatted("%H:%M")..")"..(ooc and "" or " ") or ""
+	return nut.config.get("chatShowTime") and (ooc and " " or "") .. "(" .. nut.date.getFormatted("%H:%M") .. ")" .. (ooc and "" or " ") or ""
 end
 
 -- Registers a new chat type with the information provided.
 function nut.chat.register(chatType, data)
-	if (!data.onCanHear) then
-		-- Have a substitute if the canHear property is not found.
-		data.onCanHear = function(speaker, listener)
-			-- The speaker will be heard by everyone.
-			return true
+	if (not data.onCanHear) then
+		-- Let's see first if a dynamic radius has been set.
+		if (isfunction(data.radius)) then
```

Comment:
Have you tried setting radius to a static number instead of a function? What if radius doesn't exist? Have you tested that?

Reply:
You are right but isfunction() has the ability to silently fail at **everything** except a function. So I didn't need to check if a value was present or what. This is an advantage and also a disadvantage because other functions give an error if a wrong type is entered so I guess it's still a design flaw in GLua or something.

Here are some tests:
![image](https://user-images.githubusercontent.com/26360935/118618519-ff9e4400-b7c3-11eb-93ee-2c43ddb797a5.png)

Reply:
pleas modify "radius" to also accept static numbers. This will help avoid confusion as to what radius does and does not accept

Reply:
That's fair, I'll do that.

Reply:
Done, sorry for the delay.